operator/cmd/mesh/uninstall.go

```diff
@@ -104,8 +104,8 @@ func UninstallCmd(logOpts *log.Options) *cobra.Command {
   # Uninstall all control planes and shared resources
   istioctl x uninstall --purge`,
 		Args: func(cmd *cobra.Command, args []string) error {
-			if uiArgs.revision == "" && uiArgs.filename == "" && !uiArgs.purge {
-				return fmt.Errorf("at least one of the --revision, --filename or --purge flags must be set")
+			if uiArgs.revision == "" && getRevisionFromSetFlags(uiArgs.set) == "" && uiArgs.filename == "" && !uiArgs.purge {
```

Comment:
I think you can use the following so you don't have to write your own method just to find `revision`. 
https://github.com/istio/istio/blob/96f0b401cdc7794817187334bcbee44ff1193d48/operator/pkg/manifest/shared.go#L494

Reply:
Thanks @shamsher31 
Applied your comments.